File: training/polynomial_addition/code.py
class polynomial:
	def __init__(self, values):
		self.degree = len(values)
		self.values = values

def poly_add(polA, polB):
	z = []
	posA = posB = posC = 0
	degreeA = polA.degree
	degreeB = polB.degree
	degreeC = max(degreeA, degreeB)
	while(posA <= degreeA and posB <= degreeB):
		if degreeA > degreeB:
			z.append(a.coef[posA])
			posA += 1
			degreeA -= 1
		elif degreeA == degreeB:
			z.append(polA.values[posA] + polB.values[posB])
			posA += 1
			posB += 1
			degreeA -= 1
			degreeB -= 1
		else:
			z.append(b.coef[posB])
			posB += 1
			degreeB -= 1
	return polynomial(z)

File: training/polynomial_addition/code.c
#include <stdio.h>
#define MAX_NUMBER 100

typedef struct{
	int degree;
	int coef[MAX_NUMBER];
}Polynomial;

int max_number(int a, int b) { return a > b ? a : b; }

Polynomial add_polynomial(Polynomial a, Polynomial b){
	Polynomial c;

	int posA = 0, posB = 0, posC = 0;
	int degreeA = a.degree;
	int degreeB = b.degree;
	c.degree = max_number(degreeA, degreeB);

	while(posA <= a.degree && posB <= b.degree){
		if(degreeA > degreeB){
			c.coef[posC++] = a.coef[posA++];
			degreeA--;
		}else if(degreeA == degreeB){
			c.coef[posC++] = a.coef[posA++] + b.coef[posB++];
			degreeA--;
			degreeB--;
		}else{
			c.coef[posC++] = b.coef[posB++];
			degreeB--;
		}
	}
	return c;
}

int main(){

	return 0;
}
